File: 0389-find-the-difference/0389-find-the-difference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    const map = new Map();
    for(let char of s){
        map.set(char, (map.get(char) || 0)+1);
    }
    for(let char of t){
        map.set(char, (map.get(char)-1));
        if(map.get(char) == 0) map.delete(char);
    }
    for(let [key,value] of map){
        return key;
    }
};

File: 0389-find-the-difference/0389-find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {
        Map<Character, Integer> hm = new HashMap<>();
        for(int i=0; i<t.length(); i++){
            hm.put(t.charAt(i),hm.getOrDefault(t.charAt(i),0)+1);
        }
        for(int i=0; i<s.length(); i++){
             hm.put(s.charAt(i), hm.get(s.charAt(i))-1);
             if(hm.get(s.charAt(i)) == 0) hm.remove(s.charAt(i));
        }
        for(char c : hm.keySet()){
            return c;
        }
        return 'a';
    }
}